forum/qa-plugin/block-pw/block-pw-page.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+require_once QA_INCLUDE_DIR . 'db/users.php';
+
+class block_pw_page
+{
+    private $directory;
+    private $urltoroot;
+    private $requestParts;
+
+    public function load_module(string $directory, string $urltoroot): void
+    {
+        $this->directory = $directory;
+        $this->urltoroot = $urltoroot;
+    }
+
+    public function match_request(string $request): bool
+    {
+        $this->requestParts = explode('/', $request);
+
+        return $this->requestParts[0] === 'message';
+    }
+
+    public function process_request(string $request): ?array
+    {
+        // logged in user id
+        $loggedIn = qa_get_logged_in_userid();
+        // to message user id
+        $user = $this->getUser();
+        
+        if (!$this->userExists($user)) {
+            return include QA_INCLUDE_DIR.'qa-page-not-found.php';
+        }
```

Comment:
w ogóle nie weryfikujesz strony - po co Ci w ten sposób to robić, skoro dałem Ci rozwiązanie na wyświetlenie 404 na podstawie samego requestu? Nie pytałeś bazki 2x a teraz uzależniasz ten kawałek kodu od tego czy użytkownik jest zalogowany - jasne, warto sprawdzić czy jestem zalogowany, bo w zasadzie bez tego sie nie zobaczy strony, ale też warto sprawdzić czy strona /message/fjadslfkjaslkfjaslkfdjalk istnieje i wyświetlić render stronki 404.